frontend/src/components/pages/events/EventCreator/index.tsx

```diff
@@ -0,0 +1,204 @@
+    createEvent: { event: Event };
+  }>(CREATE_EVENT, {
+    update: (cache, { data }) => {
+      data &&
+        cache.modify({
+          fields: {
+            allEvents: (existingEvents) => {
+              const newEventRef = cache.writeFragment<Event>({
+                data: data.createEvent.event,
+                fragment: gql`
+                  fragment NewEvent on Event {
+                    id
+                  }
+                `,
+              });
+              return [...existingEvents, newEventRef];
+            },
+          },
+        });
+    },
+  });
+
+  const { loading: categoryLoading, error: categoryError, data: categoryData } = useQuery(GET_CATEGORIES);
+  const { loading: userLoading, error: userError, data: userData } = useQuery<{ user: User }>(GET_USER);
+
+  if (categoryLoading || userLoading) return <CircularProgress />;
+  if (categoryError || userError) return <Typography>Det oppstod en feil.</Typography>;
+
+  if (!userData || !userData.user || !userData.user.organizations.length) {
+    router.push("/events");
+    return null;
+  }
+
+  if (!eventData.organizationId) {
+    setEventData({ ...eventData, organizationId: userData?.user.organizations[0].id });
+  }
+
+  const updateSlotDistribution = (newSlotDistribution: { category: number[]; availableSlots: number }[]) => {
+    setSlotDistribution(newSlotDistribution);
+    const usedGrades = ([] as number[])
+      .concat(...newSlotDistribution.map((dist) => dist.category))
+      .sort((a, b) => a - b);
+    setEventData({
+      ...eventData,
+      allowedGradeYears: usedGrades,
+    });
+  };
+
+  const onIsAttendableChange = (attendable: boolean) => {
+    // Reset all fields depending on isAttendable if isAttendable is disabled
+    if (attendable) {
+      setIsAttendable(true);
+      setHasSlotDistribution(false);
+    } else {
+      setEventData({
+        ...eventData,
+        availableSlots: "",
+        bindingSignup: false,
+        hasExtraInformation: false,
+        signupOpenDate: "",
+        deadline: "",
+        allowedGradeYears: [1, 2, 3, 4, 5],
+      });
+      setIsAttendable(false);
+      setHasSlotDistribution(false);
+    }
+  };
+
+  const onSubmit = () => {
+    const formattedInputData = getFormattedDataAndErrors(
+      eventData,
+      isAttendable,
+      hasSlotDistribution,
+      slotDistribution
+    );
+
+    if (formattedInputData.currentErrors.length > 0) {
+      setErrors(formattedInputData.currentErrors);
+      setOpenCreateErrorSnackbar(true);
+      return;
+    }
+    createEvent({
+      variables: {
+        eventData: formattedInputData.eventInput,
+        attendableData: isAttendable ? formattedInputData.attendableInput : undefined,
+      },
+    }).then((res) => {
+      if (res.errors) {
+        setOpenCreateErrorSnackbar(true);
+      } else {
+        setEventData(DEFAULTINPUT);
+        setOpenCreateSnackbar(true);
+        router.push("/events");
+      }
+    });
```

Comment:
This should be refactored to `onCompleted` and `onError` https://www.apollographql.com/docs/react/data/mutations/#options